Tasks/AzureFileCopy/AzureFileCopy.ps1

```diff
@@ -1,62 +1,42 @@
-    [string]$vmsAdminUserName,
-    [string]$vmsAdminPassword,
-    [string]$targetPath,
-    [string]$additionalArguments,
-    [string]$cleanTargetBeforeCopy,
-    [string]$copyFilesInParallel,
-    [string]$skipCACheck,
-    [string]$enableCopyPrerequisites,
-    [string]$outputStorageContainerSasToken,
-    [string]$outputStorageURI
-)
+[CmdletBinding()]
+param()
 
 Write-Verbose "Starting Azure File Copy Task"
-
-Write-Verbose "connectedServiceNameSelector = $connectedServiceNameSelector"
-Write-Verbose "connectedServiceName = $connectedServiceName"
-Write-Verbose "connectedServiceNameARM = $connectedServiceNameARM"
-Write-Verbose "sourcePath = $sourcePath"
-Write-Verbose "storageAccount = $storageAccount"
-Write-Verbose "storageAccountRM = $storageAccountRM"
-Write-Verbose "destination type = $destination"
-Write-Verbose "containerName = $containerName"
-Write-Verbose "blobPrefix = $blobPrefix"
-Write-Verbose "environmentName = $environmentName"
-Write-Verbose "environmentNameRM = $environmentNameRM"
-Write-Verbose "resourceFilteringMethod = $resourceFilteringMethod"
-Write-Verbose "machineNames = $machineNames"
-Write-Verbose "vmsAdminUserName = $vmsAdminUserName"
-Write-Verbose "targetPath = $targetPath"
-Write-Verbose "additionalArguments = $additionalArguments"
-Write-Verbose "cleanTargetBeforeCopy = $cleanTargetBeforeCopy"
-Write-Verbose "copyFilesInParallel = $copyFilesInParallel"
-Write-Verbose "skipCACheck = $skipCACheck"
-Write-Verbose "enableCopyPrerequisites = $enableCopyPrerequisites"
-Write-Verbose "outputStorageContainerSASToken = $outputStorageContainerSASToken"
-Write-Verbose "outputStorageURI = $outputStorageURI"
-
-if ($connectedServiceNameSelector -eq "ConnectedServiceNameARM")
+Trace-VstsEnteringInvocation $MyInvocation
+
+# Get inputs for the task
+$ConnectedServiceNameSelector = Get-VstsInput -Name ConnectedServiceNameSelector -Require
+$SourcePath = Get-VstsInput -Name SourcePath -Require
+$Destination = Get-VstsInput -Name Destination -Require
+$ConnectedServiceName = Get-VstsInput -Name ConnectedServiceName
+$ConnectedServiceNameARM = Get-VstsInput -Name ConnectedServiceNameARM
+$StorageAccount = Get-VstsInput -Name StorageAccount
+$StorageAccountRM = Get-VstsInput -Name StorageAccountRM
+$ContainerName = Get-VstsInput -Name ContainerName
+$BlobPrefix = Get-VstsInput -Name BlobPrefix
+$EnvironmentName = Get-VstsInput -Name EnvironmentName
+$EnvironmentNameRM = Get-VstsInput -Name EnvironmentNameRM
+$ResourceFilteringMethod = Get-VstsInput -Name ResourceFilteringMethod
+$MachineNames = Get-VstsInput -Name MachineNames
+$VmsAdminUserName = Get-VstsInput -Name VmsAdminUsername
+$VmsAdminPassword = Get-VstsInput -Name VmsAdminPassword
+$TargetPath = Get-VstsInput -Name TargetPath
+$AdditionalArguments = Get-VstsInput -Name AdditionalArguments
+$CleanTargetBeforeCopy = Get-VstsInput -Name CleanTargetBeforeCopy
+$CopyFilesInParallel = Get-VstsInput -Name CopyFilesInParallel
+$SkipCACheck = Get-VstsInput -Name SkipCACheck
```

Comment:
You can use  Get-VstsInput -Name <NAME> -AsBool for specifying boolean parameters